packages/theme-sections/theme-sections.js

```diff
@@ -212,13 +212,16 @@ function normalizeContainers(containers) {
 }
 
 if (window.Shopify.designMode) {
-  document.addEventListener('shopify:section:load', function(event) {
-    var id = event.detail.sectionId;
-    var container = event.target.querySelector(
-      '[' + SECTION_ID_ATTR + '="' + id + '}"]'
+  document.addEventListener('shopify:section:load', event => {
+    const container = event.target;
+    const containerDataset = container.dataset;
+
+    // Hack solution - pulling the first data-attribute since the identifier is unknown
+    const sectionData = JSON.parse(
```

Comment:
Changing the way we access data attributes should be in a separate PR as it's a different problem than the bug of reloading the section. #36 is a more direct fix.